rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java

```diff
@@ -354,33 +385,48 @@ protected void buildForm() {
         this.window.centerInPage();
     }
 
-    private void prepareCreateAndDeployFormAndSubmit(VLayout nodeSourceWindowLayout,
-            Label nodeSourcePluginsWaitingLabel, DynamicForm nodeSourcePluginsForm, Label nodeSourceWindowLabel,
-            TextItem nodeSourceNameItem, CheckboxItem nodesRecoverableItem, List<IButton> buttonList,
-            boolean nodeSourceEdited) {
+    private void applyModificationsToNodeSource(VLayout nodeSourceWindowLayout, Label nodeSourcePluginsWaitingLabel,
+            DynamicForm nodeSourcePluginsForm, Label nodeSourceWindowLabel, TextItem nodeSourceNameItem,
+            CheckboxItem nodesRecoverableItem, List<IButton> buttonList, NodeSourceAction nodeSourceAction) {
+
+        nodeSourcePluginsForm.setValue(DEPLOY_FORM_KEY, Boolean.FALSE.toString());
+
+        saveNodeSource(nodeSourceWindowLayout,
+                       nodeSourcePluginsWaitingLabel,
+                       nodeSourcePluginsForm,
+                       nodeSourceWindowLabel,
+                       nodeSourceNameItem,
+                       nodesRecoverableItem,
+                       buttonList,
+                       nodeSourceAction);
+    }
+
+    private void saveAndDeployNodeSource(VLayout nodeSourceWindowLayout, Label nodeSourcePluginsWaitingLabel,
+            DynamicForm nodeSourcePluginsForm, Label nodeSourceWindowLabel, TextItem nodeSourceNameItem,
+            CheckboxItem nodesRecoverableItem, List<IButton> buttonList, NodeSourceAction nodeSourceAction) {
 
         nodeSourcePluginsForm.setValue(DEPLOY_FORM_KEY, Boolean.TRUE.toString());
 
-        prepareCreateOnlyFormAndSubmit(nodeSourceWindowLayout,
-                                       nodeSourcePluginsWaitingLabel,
-                                       nodeSourcePluginsForm,
-                                       nodeSourceWindowLabel,
-                                       nodeSourceNameItem,
-                                       nodesRecoverableItem,
-                                       buttonList,
-                                       nodeSourceEdited);
+        saveNodeSource(nodeSourceWindowLayout,
+                       nodeSourcePluginsWaitingLabel,
+                       nodeSourcePluginsForm,
+                       nodeSourceWindowLabel,
+                       nodeSourceNameItem,
+                       nodesRecoverableItem,
+                       buttonList,
+                       nodeSourceAction);
     }
 
-    private void prepareCreateOnlyFormAndSubmit(VLayout nodeSourceWindowLayout, Label nodeSourcePluginsWaitingLabel,
+    private void saveNodeSource(VLayout nodeSourceWindowLayout, Label nodeSourcePluginsWaitingLabel,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)